File: expectanim/src/main/java/com/github/florent37/expectanim/core/position/PositionAnimExpectationLeftOfParent.java
package com.github.florent37.expectanim.core.position;

import android.view.View;

/**
 * Created by florentchampigny on 17/02/2017.
 */

public class PositionAnimExpectationLeftOfParent extends PositionAnimExpectation {

    public PositionAnimExpectationLeftOfParent() {
        setForPositionX(true);
    }

    @Override
    public Float getCalculatedValueX(View viewToMove) {
        return getMargin(viewToMove);
    }

    @Override
    public Float getCalculatedValueY(View viewToMove) {
        return null;
    }
}

File: expectanim/src/main/java/com/github/florent37/expectanim/core/position/PositionAnimExpectationLeftOfParent.kt
package com.github.florent37.expectanim.core.position

import android.view.View

class PositionAnimExpectationLeftOfParent : PositionAnimExpectation() {
    init {
        isForPositionX = true
    }

    override fun getCalculatedValueX(viewToMove: View): Float? {
        return getMargin(viewToMove)
    }

    override fun getCalculatedValueY(viewToMove: View): Float? {
        return null
    }
}
